### PR TITLE
Ignore macros for case when the name provided in used macros place

### PR DESCRIPTION
**Pull request by vkatsuba**
Related to https://github.com/AdRoll/rebar3_hank/issues/117.
**Test Result:** https://github.com/vkatsuba/rebar3_hank/runs/2500838161

### CODE REVIEW

**Comment by elbrujohalcon** [comment]
Ohhhh! Now I see what you mean, @vkatsuba! It's not that the macro was unparseable… The issue is that macros with "arity" `none` can be used as function calls (i.e. with _any_ arity).

**Comment by vkatsuba** [comment]
> Ohhhh! Now I see what you mean, @vkatsuba! It's not that the macro was unparseable… The issue is that macros with "arity" `none` can be used as function calls (i.e. with _any_ arity).

Yep.

File: src/rules/unused_macros.erl [hunk 1]
```diff
@@ -78,3 +78,10 @@ ignored({Name, _Arity}, Name) ->
     true;
 ignored(_Pattern, _IgnoreSpec) ->
     false.
+
+is_member(_, []) ->
+    false;
+is_member(H, [{H, _} | _]) ->
+    true;
+is_member(H, [_ | T]) ->
+    is_member(H, T).
```
Comment (elbrujohalcon) [inline src/rules/unused_macros.erl:87]: Would this work if we make it like this…

```suggestion
is_member(MacroName, none, UsedMacros) ->
    lists:keymember(MacroName, 1, UsedMacros);
is_member(MacroName, MacroArity, UsedMacros) ->
    lists:member({MacroName, MacroArity}, UsedMacros).
```

…or like this…

```suggestion
is_member({MacroName, none}, UsedMacros) ->
    lists:keymember(MacroName, 1, UsedMacros);
is_member(Macro, UsedMacros) ->
    lists:member(Macro, UsedMacros).
```

…?

I mean… only macros with `none` arity can be used as function calls.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: src/rules/unused_macros.erl (hunk 1 above)
Comment (pbrudnick) [reply src/rules/unused_macros.erl:87]: I agree with this, checking `none` or the "old way"

**Comment by pbrudnick** [comment]
Good catch @vkatsuba !

File: src/rules/unused_macros.erl (hunk 1 above)
Comment (vkatsuba) [reply src/rules/unused_macros.erl]: Done

**Review by elbrujohalcon — APPROVED** [review]

**Review by pbrudnick — APPROVED** [review]

**Pull request merged**